### PR TITLE
add coordinate criteria table in docs

### LINKED ISSUE #138: Document coordinate criteria.
Ideally we should have a script that parses the `coordinate_criteria` dictionary and outputs a table in `.rst` format that would get rendered during the docs build. Don't know how feasible this is :)

### PR DESCRIPTION
**Pull request by malmans2**
Close #138: I'd also find it very useful so I gave it a go.

I've added a script to create a table with coordinate criteria in the documentation.
Not sure if I placed the table and the script in the right place (maybe the *script should just go in `doc/`?), but that should be easy to change.

Right now it creates this.

attribute_table

### CODE REVIEW

File: doc/contributing.rst [hunk 1]
```diff
@@ -32,6 +32,11 @@ This dictionary contains criteria for identifying axis and coords using CF attri
 
    ~accessor.coordinate_criteria
 
+.. csv-table::
+   :file: _build/csv/coordinate_criteria.csv
```
Comment (malmans2) [inline doc/contributing.rst:36]: pre-commit is failing because this doesn't exist, it's created by `conf.py`

Comment (dcherian) [reply doc/contributing.rst:36]: I think we add `csv-table` to `ignore_directives` at the end of `setup.cfg`

**Comment by dcherian** [comment]
Thanks @malmans2 this is a great improvement!

My only suggestion is to make the table more prominent. Can you put it in the top bar so it's easy to find? We can rearrange things later (maybe we want the top bar to say "Reference" and then have "API reference" and "Coordinate Criteria" as subsections)

**Comment by dcherian** [comment]
It would be nice to add a link to that `introduction.ipynb` notebook

**Comment by malmans2** [comment]
What about CF Criteria in the top bar, splitting axes and coordinates?

top_bar

**Comment by dcherian** [comment]
That looks nice. How about transposing the table since number of rows < number of columns?

**Comment by dcherian** [comment]
Should also a whats-new note for this awesome change.!

**Comment by dcherian** [comment]
Could also add a table of regex patterns used for guessing

**Comment by malmans2** [comment]
Regex table added!

regex

**Comment by dcherian** [comment]
This is really great. Thanks!

**Pull request merged**